gogen/codegen.go

```diff
@@ -0,0 +1,429 @@
+// Package gogen is a library for generating Go structs from a YANG schema.
+package gogen
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/openconfig/ygot/internal/igenutil"
+	"github.com/openconfig/ygot/util"
+	"github.com/openconfig/ygot/ygen"
+	"github.com/openconfig/ygot/ygot"
+)
+
+// GoCodeGenerator is a structure that is used to pass arguments as to
+// how the output Go code should be generated.
+type GoCodeGenerator struct {
+	// Config stores the configuration parameters used for code generation.
+	Config ygen.GeneratorConfig
+	// GoOptions stores a struct which stores Go code generation specific
+	// options for the code generaton.
+	GoOptions GoOpts
+}
+
+// GoOpts stores Go specific options for the code generation library.
+type GoOpts struct {
```

Comment:
```suggestion
type Opts struct {
```

Reply:
This is one place where I wasn't convinced of removing the "Go" qualifier. The reason is `ygen.GeneratorConfig` also configures some generation options such as CompressBehaviour, the enum options (e.g. `SkipEnumDeduplication`) and others like `ExcludeModules` and `GenerateJSONSchema` (maybe this one should go into GoOpts), so I think calling it `GenOptions` could be confusing since users might assume it contains all of the generation options when in reality it only contains the Go-specific ones. What do you think?

Current `CodeGenerator` definition containing all generation options.
```go
// CodeGenerator is a structure that is used to pass arguments as to                                                                                                                                                                                                                                  
// how the output Go code should be generated.                                                                                                                                                                                                                                                        
type CodeGenerator struct {                                                                                                                                                                                                                                                                           
        // Config stores the configuration parameters used for code generation.                                                                                                                                                                                                                       
        Config ygen.GeneratorConfig                                                                                                                                                                                                                                                                   
        // GoOptions stores a struct which stores Go code generation specific                                                                                                                                                                                                                         
        // options for the code generaton.                                                                                                                                                                                                                                                            
        GoOptions GoOpts                                                                                                                                                                                                                                                                              
}                                                                                                                                                                                                                                                                                                     
```

`Config` seems like a bad name. Maybe `BasicConfig` or something is better?

Reply:
Proposal to address this here: https://github.com/openconfig/ygot/pull/706 PTAL

Reply:
Thanks, that looks a good solution to the problem. I'll resolve this thread.